### ISSUE TITLE
Duplicate results from racer

### ISSUE THREAD

**Issue by Wilfred** [racer-rust/racer#645]
Given the code:

```rust
pub enum Level {
}
```

Racer gives:

```
$ CARGO_HOME=/home/wilfred/.cargo RUST_SRC_PATH=/home/wilfred/src/rustc-1.10.0/src /home/wilfred/.cargo/bin/racer complete-with-snippet 1 14 /home/wilfred/projects/bfc/src/diagnostics.rs
PREFIX 9,14,Level
MATCH Level;Level;1;9;/home/wilfred/projects/bfc/src/diagnostics.rs;Enum;pub enum Level {;""
MATCH Level;Level;1;9;/home/wilfred/projects/bfc/src/diagnostics.rs;Enum;pub enum Level {;""
END
```

**Comment by TedDriggs** [comment]
I've seen suggestions appear 5 or 6 times over.

**Comment by andjo403** [comment]
I can not reproduce this fault on the latest commit on master
commit fbfa038
if there is no other exempels when there is duplicated results I think that this can be closed.

**Comment by TedDriggs** [comment]
I'm still able to reproduce this; I see the same module appearing 3 times in the autocomplete suggestions. Is it possible this is related to using a glob import such as `use super::*;`?

**Comment by jwilm** [comment]
Thanks @TedDriggs. Are they results you see _identical_? Maybe we can just hack around this by filtering dupes if they are 100% the same.

**Comment by TedDriggs** [comment]
Yes, they appear to be identical (all 3 show the exact same path when highlighted)

![screen shot 2017-01-04 at 2 13 39 pm](https://cloud.githubusercontent.com/assets/4805575/21661244/0a45e168-d288-11e6-9868-a60f3611d134.png)

**Comment by TedDriggs** [comment]
I found and addressed one source of duplication in #774, so methods should no longer appear multiple times. However, I still see duplicate mod results when completing in the project I repro'd this in back in January, so I'll dig deeper into that.
